http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java

```diff
@@ -415,6 +415,7 @@ private String expandOrEncode(String value, Map<String, ? super Object> values)
     }
 
     private String encode(String userInfo) {
-        return URLEncoder.encode(userInfo, StandardCharsets.UTF_8);
+        return URLEncoder.encode(userInfo, StandardCharsets.UTF_8)
+                         .replaceAll("\\+", "%20"); // to match RFC3986
```

Comment:
You can't just change the framework's behavior. I think the most reasonable solution is to add a setting - how to encode a space. Then you leave backward compatibility and whoever needs it will enable RFC support mode

Reply:
Agree with you Im gonna add it

Reply:
Agree with you Im gonna add it

Reply:
It's not a good look for the framework to be handling this incorrectly when competing frameworks get it right. I'd argue it should be changed to the correct behavior, with an option to revert to the prior (incorrect) encoding behavior. The Apache URIBuilder I mentioned in my earlier comment gets this right, defaulting to the correct encoding of `%20` but providing an optional argument for encoding spaces as `+` if needed.

Reply:
@yawkat need your final decision

Reply:
Indeed we can't change it this way, at least not without a config option.